saleor/order/tasks.py

```diff
@@ -21,3 +32,58 @@ def send_order_updated(order_ids):
     manager = get_plugins_manager()
     for order in Order.objects.filter(id__in=order_ids):
         manager.order_updated(order)
+
+
+def _batch_ids(iterable, batch_size=1):
+    length = len(iterable)
+    for index in range(0, length, batch_size):
+        yield iterable[index : min(index + batch_size, length)]
+
+
+def _queryset_in_batches(queryset):
+    ids = queryset.values_list("id", flat=True)
+    for ids_batch in _batch_ids(ids, 1000):
+        yield ids_batch
+
+
+def _expire_orders(manager, now):
+    qs = Order.objects.filter(
+        status=OrderStatus.UNCONFIRMED,
+        channel__expire_orders_after__isnull=False,
+        total_charged_amount=Decimal(0),
+    ).select_related("channel")
+    for ids_batch in _queryset_in_batches(qs):
+        with traced_atomic_transaction():
+            orders = Order.objects.filter(id__in=ids_batch).prefetch_related("user")
+            orders_to_update = []
+            for order in orders:
+                if _process_order(order, manager, now):
+                    orders_to_update.append(order)
+            Order.objects.bulk_update(orders_to_update, ["status", "updated_at"])
+
+
+def _process_order(order, manager, now):
+    if (
+        order.channel.expire_orders_after is not None
+        and order.created_at
```

Comment:
Why we can't filter it out during the fetching all expired orders? `qs` from line 50.  With current approach and big amount of `expire_after` this will cause that we will need to fetch a lot of unneeded orders.

Reply:
Thanks to it, you don't need this method, you can just call `expire_orders` to all fetched orders.